### ISSUE TITLE
flux delete helmrelease does not work unless followed by flux suspend

### ISSUE THREAD

**Issue by emily-zall** [fluxcd/flux2#4507]
### Describe the bug

I run `flux delete helmrelease` and it says the helmrelease was deleted but it is still on the list. I want to delete it so it can be recreated. 

```
➜ flux delete helmrelease simbachain-blocks-ks18 -n blocks-simba-ks18
Are you sure you want to delete this helmrelease: y
► deleting helmrelease simbachain-blocks-ks18 in blocks-simba-ks18 namespace
✔ helmrelease deleted

➜ flux get helmreleases -n blocks-simba-ks18
NAME                  	REVISION	SUSPENDED	READY	MESSAGE
blocks-simba-ks18     	0.1.107 	False    	True 	Release reconciliation succeeded
neo4j-h18-test        	5.11.0  	False    	True 	Release reconciliation succeeded
simbachain-blocks-ks18	        	False    	False	Helm install failed: context deadline exceeded
```

*(workaround)* I noticed that several minutes later when I ran flux suspend, then it deleted for real
```
➜ flux suspend helmrelease simbachain-blocks-ks18 -n blocks-simba-ks18
► suspending helmrelease simbachain-blocks-ks18 in blocks-simba-ks18 namespace
✔ helmrelease suspended
```

### Steps to reproduce

```
➜ flux delete helmrelease simbachain-blocks-ks18 -n blocks-simba-ks18
Are you sure you want to delete this helmrelease: y
► deleting helmrelease simbachain-blocks-ks18 in blocks-simba-ks18 namespace
✔ helmrelease deleted

➜ flux get helmreleases -n blocks-simba-ks18
NAME                  	REVISION	SUSPENDED	READY	MESSAGE
blocks-simba-ks18     	0.1.107 	False    	True 	Release reconciliation succeeded
neo4j-h18-test        	5.11.0  	False    	True 	Release reconciliation succeeded
simbachain-blocks-ks18	        	False    	False	Helm install failed: context deadline exceeded
```

### Expected behavior

helmrelease is deleted

### OS / Distro

Mac OS 13.4.1

### Flux version

v0.41.2

### Flux check

➜ flux check
► checking prerequisites
✗ flux 0.41.2 <2.2.2 (new version is available, please upgrade)
✔ Kubernetes 1.26.6 >=1.20.6-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.36.1
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.36.1
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.30.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v1.1.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v1.1.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v1.1.1
► checking crds
✔ alerts.notification.toolkit.fluxcd.io/v1beta2
✔ buckets.source.toolkit.fluxcd.io/v1beta2
✔ gitrepositories.source.toolkit.fluxcd.io/v1
✔ helmcharts.source.toolkit.fluxcd.io/v1beta2
✔ helmreleases.helm.toolkit.fluxcd.io/v2beta1
✔ helmrepositories.source.toolkit.fluxcd.io/v1beta2
✔ imagepolicies.image.toolkit.fluxcd.io/v1beta2
✔ imagerepositories.image.toolkit.fluxcd.io/v1beta2
✔ imageupdateautomations.image.toolkit.fluxcd.io/v1beta1
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2
✔ providers.notification.toolkit.fluxcd.io/v1beta2
✔ receivers.notification.toolkit.fluxcd.io/v1
✔ all checks passed

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hiddeco** [comment]
You are reporting an issue with a version that’s relatively old. Please confirm you can reproduce this with the latest version, as I suspect this will not be the case.

**Comment by emily-zall** [comment]
resolved after upgrading flux cli to 2.2.2
